python/pyspark/mllib/common.py

```diff
@@ -130,20 +130,39 @@ def callMLlibFunc(name, *args):
     return callJavaFunc(sc, api, *args)
 
 
-class JavaModelWrapper(object):
+class JavaCallable(object):
     """
-    Wrapper for the model in JVM
+    Wrapper for an object in JVM to make Java calls
     """
-    def __init__(self, java_model):
-        self._sc = SparkContext.getOrCreate()
-        self._java_model = java_model
+    def __init__(self, sc, java_obj):
+        self._sc = sc
+        self._java_obj = java_obj
 
     def __del__(self):
-        self._sc._gateway.detach(self._java_model)
+        self._sc._gateway.detach(self._java_obj)
+
+    @classmethod
+    def fromActiveSparkContext(cls, java_obj):
+        """Create from a currently active context"""
+        sc = SparkContext._active_spark_context
+        return cls(sc, java_obj)
 
     def call(self, name, *a):
```

Comment:
this ends up in the api docs because ML has the `:inherited-members:` flag on [here](https://raw.githubusercontent.com/apache/spark/master/python/docs/pyspark.ml.rst)

Is there any way to avoid this besides making these private?